File: app/src/main/java/com/agrogames/islandsofwar/graphics/impl/gl/GLSurfaceView.java
package com.agrogames.islandsofwar.graphics.impl.gl;
import android.content.Context;
import android.view.MotionEvent;
import android.view.View;

import com.agrogames.islandsofwar.graphics.abs.RenderManager;

public class GLSurfaceView extends android.opengl.GLSurfaceView {

    public GLSurfaceView(Context context, RenderManager manager){
        super(context);

        setEGLContextClientVersion(2);

        GLRenderer renderer = new GLRenderer(context, manager);
        setOnTouchListener((v, e) -> {
            v.performClick();
            if(e.getAction() == MotionEvent.ACTION_UP){
                renderer.onTouch(e.getX(), e.getY());
            } else if(e.getAction() == MotionEvent.ACTION_MOVE){
                if(e.getPointerCount() == 1){
                    renderer.onMove(e.getX(), e.getY());
                } else{
                    renderer.onZoom(e.getX(0), e.getY(0), e.getX(1), e.getY(1));
                }
            }
            return true;
        });

        setRenderer(renderer);
    }
}

File: app/src/main/java/com/agrogames/islandsofwar/graphics/gl/GLSurfaceView.kt
package com.agrogames.islandsofwar.graphics.gl

/*
@SuppressLint("ViewConstructor")
class GLSurfaceView(context: Context, manager: RenderManager) : GLSurfaceView(context) {
    private val renderer: GLRenderer

    init {
        setEGLContextClientVersion(2)
        renderer = GLRenderer(context, manager)
        setOnTouchListener { v: View, e: MotionEvent ->
            v.performClick()
            if (e.action == MotionEvent.ACTION_UP) {
                renderer.onTouch(e.x, e.y)
            } else if (e.action == MotionEvent.ACTION_MOVE) {
                if (e.pointerCount == 1) {
                    renderer.onMove(e.x, e.y)
                } else {
                    renderer.onZoom(e.getX(0), e.getY(0), e.getX(1), e.getY(1))
                }
            }
            true
        }
        setRenderer(renderer)
    }

    fun setManager(m: RenderManager) {
        renderer.setManager(m)
    }
}*/